### PR TITLE
Use FlexAmps to start charge

### PR DESCRIPTION
**Pull request by dschuesae**
Have a look at this. It works for my usecase - but perhaps I've overseen something.

### CODE REVIEW

**Comment by MikeBishop** [comment]
I'm curious - why would you do this instead of putting in a generation offset? The intent of this feature was to "ride out" dips in production so you don't cycle the contactor needlessly. The code looks reasonable, but I don't understand the use-case.

**Comment by dschuesae** [comment]
It's cheaper to have a mixed power price and not to sell it, when your offset is below 6A (3 phases) until a specific limit. And at night I have a better price to buy power. So it's best to to take the power as long as I produce it by myself, then as long mixed price is better than I send to the grid and then at night. (No battery at moment).

File: lib/TWCManager/TWCSlave.py [hunk 1]
```diff
@@ -65,10 +67,16 @@ class TWCSlave:
 
     def __init__(self, TWCID, maxAmps, config, master):
         self.config = config
+        try:
+            self.configConfig = config["config"]
+        except KeyError:
+            self.configConfig = {}
```
Comment (MikeBishop) [inline lib/TWCManager/TWCSlave.py:73]:
```suggestion
        self.configConfig.get("config", {})
```

File: lib/TWCManager/TWCSlave.py [hunk 2]
```diff
@@ -594,18 +604,19 @@ def receive_slave_heartbeat(self, heartbeatData):
...
         if desiredAmpsOffered < minAmpsToOffer:
             self.master.debugLog(
                 10,
                 "TWCSlave  ",
-                "desiredAmpsOffered ("
+                "desiredAmpsOffered: "
                 + str(desiredAmpsOffered)
-                + ") < minAmpsToOffer:"
-                + str(minAmpsToOffer),
+                + " < minAmpsToOffer:"
```
Comment (MikeBishop) [inline lib/TWCManager/TWCSlave.py:617]:
```suggestion
                + " < minAmpsToOffer: "
```

File: lib/TWCManager/TWCSlave.py [hunk 3]
```diff
@@ -65,10 +67,16 @@ class TWCSlave:
...
         self.config = config
+        try:
+            self.configConfig = config["config"]
+        except KeyError:
+            self.configConfig = {}
         self.master = master
         self.TWCID = TWCID
         self.maxAmps = maxAmps
-        self.wiringMaxAmps = config["config"]["wiringMaxAmpsPerTWC"]
+
```
Comment (MikeBishop) [inline lib/TWCManager/TWCSlave.py:77]:
```suggestion
```

**Comment by dschuesae** [comment]
It's little bit complicated to explain. 

Best and lowest price is to charge full solar --> 0$. If I need more, it's better to charge at night, than on day (0.08$ difference) -->0.21$. But it is even better to charge at daylight even if solar power is too, because if I don't use this power - I get not really much money (0.02$) - and this power I have to pay during night. (between 0 and 0.19$).

And as my car is only at home on weekends at daylight I stop charging at night for friday and saturday (even thursday - if don't need the car a lot) - but I charge on sunday night if still necessary. 

Best is - I get 0.25$ from every kwh for the energy needed from my company... 

I hope this helps to understand.

**Comment by MikeBishop** [comment]
Ah, I think I see.  Because you have no batteries and your return exporting is poor, it's better on the whole to import some fraction of your charging power than ever export anything.  Not an application of flex amps I had thought about, but definitely reasonable.  Thanks for explaining!

**Pull request merged**